toolbox/githubctl/main.go

```diff
@@ -45,14 +60,111 @@ func fastForward(repo, baseBranch, refSHA *string) error {
 	return githubClnt.FastForward(*repo, *baseBranch, *refSHA)
 }
 
+func createIstioReleaseTag() error {
+	assertNotEmpty("base_branch", baseBranch)
+	pickledDeps, err := githubClnt.GetFileContent(istioRepo, *baseBranch, istioDepsFile)
+	if err != nil {
+		return err
+	}
+	deps, err := u.DeserializeDepsFromString(pickledDeps)
+	if err != nil {
+		return err
+	}
+	releaseTag, err := githubClnt.GetFileContent(istioRepo, *baseBranch, releaseTagFile)
+	if err != nil {
+		return err
+	}
+	releaseTag = strings.Trim(releaseTag, "\n")
+	releaseMsg := "Istio Release " + releaseTag
+	for _, dep := range deps {
+		if err := githubClnt.CreateAnnotatedTag(
+			dep.RepoName, releaseTag, dep.LastStableSHA, releaseMsg); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func findPathToArchives(consoleLog *string) (string, error) {
+	lines := strings.Split(*consoleLog, "\n")
+	lastLine := lines[len(lines)-2]
+	if strings.HasPrefix(lastLine, logPrefix) {
+		dir := lastLine[len(logPrefix):]
+		return dir, nil
+	}
+	return "", fmt.Errorf("failed to find path to release archives from console log")
+}
+
+func updateIstioVersion(releaseTag string) error {
+	hubCommaTag := fmt.Sprintf("%s,%s", dockerHub, releaseTag)
+	istioctl := fmt.Sprintf(
+		"https://storage.googleapis.com/istio-artifacts/pilot/%s/artifacts/istioctl", releaseTag)
+	cmd := fmt.Sprintf("./install/updateVersion.sh -p %s -c %s -x %s -i %s",
+		hubCommaTag, hubCommaTag, hubCommaTag, istioctl)
+	_, err := u.Shell(cmd)
+	return err
+}
+
+func createIstioReleaseUploadArtifacts() error {
+	assertNotEmpty("base_branch", baseBranch)
+	assertNotEmpty("next_release", nextRelease)
+	if err := u.CloneRepoCheckoutBranch(
+		githubClnt, istioRepo, *baseBranch); err != nil {
+		return err
+	}
+	defer func() {
+		if err := u.RemoveLocalRepo(istioRepo); err != nil {
+			log.Fatalf("Error during clean up: %v\n", err)
+		}
+	}()
+	output, err := u.Shell("bash ./release/create_release_archives.sh")
```

Comment:
Please update create_release_archives to take in a flag for where to generate artifacts.